### ISSUE TITLE
Should we consider separating the instruction decode and execute phases?

### ISSUE THREAD

**Issue by virinci** [Arka-Mondal/riscvemu#2]
It might make testing each part independently a bit easier.

**Comment by Arka-Mondal** [comment]
Yeah, that will be better ig. What are you ideas with this? What functions can be exposed to the outer world from `execute` translation layer.
